browser/js/common/directives/search-bar/search-bar.directive.js

```diff
@@ -0,0 +1,76 @@
+app.directive( 'searchBar', function ( $rootScope, AuthService, AUTH_EVENTS, $state ) {
+  return {
+    controller: 'TypeaheadCtrl',
+    template: searchBarTemplate
+  }
+} );
+
+app.controller( 'TypeaheadCtrl', function ( $scope, $http, $state, $rootScope ) {
+
+  var _selected;
+
+/**
+ * [getSongs sends a query to the search route]
+ * @param  {String} searchQuery [title string to search]
+ * @return {Object[]}             [an array of song objects]
+ */
+  $scope.getSongs = function ( searchQuery ) {
+    console.log( searchQuery );
+    let config = {
+      params: {
+        where: {
+          title: {
+            like: '%' + searchQuery + '%'
+          }
+        }
+      }
+    }
+    return searchQuery ? $http.get( '/api/v1/songs/s', config )
+      .then( response => response.data ) : [];
+  };
+
+  $scope.updateSongs = function ( customSelected ) {
+    $scope.getSongs( customSelected )
+      .then( songs => $scope.songs = songs );
```

Comment:
Getting the error '$scope.getSongs.then() is not a function' because getSongs doesn't return a promise.